apis/clouddms/v1alpha1/connectionprofile_identity.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"fmt"
+	"strings"
+)
+
+type ConnectionProfileIdentity struct {
+	Project           string
+	Location          string
+	ConnectionProfile string
+}
+
+func (i *ConnectionProfileIdentity) FromExternal(external string) error {
+	tokens := strings.Split(external, "/")
+	if len(tokens) != 6 {
+		return fmt.Errorf("invalid external format: %q, expected projects/{project}/locations/{location}/connectionProfiles/{connectionProfile}", external)
+	}
+	if tokens[0] != "projects" {
+		return fmt.Errorf("invalid external format: %q, expected projects segment", external)
```

Comment:
This error message is sort of cryptic, which is why I've used this pattern:

```
if len(tokens) == 6 && tokens[0] == "projects" && ... {
  ...
  return nil
}
return fmt.Errorf("invalid external format: %q, expected projects/{project}/locations/{location}/connectionProfiles/{connectionProfile}", external)
```

Also, you were previously defining the expected format in a constant called something like ConnectionProfileIdentityURLFormat, which I think is a nice pattern